### PR TITLE
Cindy/location

### PR DESCRIPTION
**Pull request by cindyxue**
Add map and location to user profile page.

### CODE REVIEW

File: src/main/webapp/js/map.js [hunk 1]
```diff
@@ -4,17 +4,47 @@ let map;
  * Build the map
  */
 function createMap(){
-    map = new google.maps.Map(document.getElementById('map'), {
-      center: {lat: 37.422, lng: -122.084},
-      zoom: 16
-    });
+    // map = new google.maps.Map(document.getElementById('map'), {
```
Comment (emto) [inline src/main/webapp/js/map.js]: delete comment?

File: src/main/webapp/js/map.js [hunk 2]
```diff
@@ -4,17 +4,47 @@ let map;
...
+        zoom: 16
+      });
+
+      // Browser doesn't support Geolocation
+      handleLocationError(false, infoWindow, map.getCenter());
+    }
 
     // Add a marker at Googleplex
-    addMarker('Stan the T-Rex', 37.421903, -122.084674, null, 'This is Stan, the T-Rex statue.', map);
+    // addMarker('Stan the T-Rex', 37.421903, -122.084674, null, 'This is Stan, the T-Rex statue.', map);
```
Comment (emto) [inline src/main/webapp/js/map.js]: delete comment

File: src/main/webapp/js/map.js [hunk 3]
```diff
@@ -4,17 +4,47 @@ let map;
...
 
     // Add a marker for the dog park
     var image = 'https://developers.google.com/maps/documentation/javascript/examples/full/images/beachflag.png';
-    addMarker('Stan the T-Rex', 37.428352, -122.077574, image, 'This is a park for dogs.', map);
+    // addMarker('Stan the T-Rex', 37.428352, -122.077574, image, 'This is a park for dogs.', map);
```
Comment (emto) [inline src/main/webapp/js/map.js]: delete comment

File: src/main/webapp/js/profile-map.js [hunk 4]
```diff
@@ -0,0 +1,189 @@
+let map;
+let editMarker;
+
+/**
+ * Build the map
+ */
+function createMap(){
+    // map = new google.maps.Map(document.getElementById('map'), {
```
Comment (emto) [inline src/main/webapp/js/profile-map.js]: comment??

File: src/main/webapp/js/profile-map.js [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+      handleLocationError(false, infoWindow, map.getCenter());
+
+      map.addListener('click', (event) => {
+        createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+      });
+
+      fetchMarkers();
+    }
+
+    // map.addListener('click', (event) => {
```
Comment (emto) [inline src/main/webapp/js/profile-map.js]: +1

**Review by minhphu20 — APPROVED** [review]

**Pull request merged**